### PR TITLE
reset login data when view disappear:

### LINKED ISSUE #125: Username and Password on Login screen does not reset after login
**Describe the bug**
The username and the password of the user used for login remains preserved even after a successful login. Ideally, it should be reset.

**To Reproduce**
1. Login using username/email & password
2. Logout
3. See the username and password on the login screen

**Expected behavior**
The username and password should reset on a successful login and the text fields on login screen should be empty when a user logs out.

**Screenshots**
Username and password present on login screen appears after signing out from settings.
image

**Smartphone (please complete the following information):**
 - Device: iPhone
 - OS: iOS 13.6
 - Xcode 11.6

### PR DESCRIPTION
**Pull request by iMostfa**
### Description
 when login view is going to disappear, login data we should reset the login data, to avoid the appearance for it again when the user logs out, as a fix for this [bug](https://github.com/anitab-org/mentorship-ios/issues/125)

result 
 

Fixes #125 

### Type of Change:
**Delete irrelevant options.**

- Code
 - User Interface
 
**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
all tests passed, it doesn't affect them. 

### Checklist:
**Delete irrelevant options.**

 
**Code/Quality Assurance Only**
 - [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sunjunkie — DISMISSED** [review]
Looks good. Thank you!

File: mentorship ios/ViewModels/LoginViewModel.swift [hunk 1]
```diff
@@ -30,5 +30,11 @@ class LoginViewModel: ObservableObject {
     func attemptAppleLogin() {
         appleSignInCoordinator.handleAuthorizationAppleIDButtonPress()
     }
+
+    //reset tthe login data when the view is going to disappear
+    //fix for:https://github.com/anitab-org/mentorship-ios/issues/125
```
Comment (yugantarjain) [inline mentorship ios/ViewModels/LoginViewModel.swift]: Line 34: typo (also, can be made more explicit in intention, example: `used to reset the login data on a successful login`)
Line 35: not really required

File: mentorship ios/ViewModels/LoginViewModel.swift [hunk 2]
```diff
@@ -30,5 +30,11 @@ class LoginViewModel: ObservableObject {
...
+    func resetLogin() {
+        self.loginData = .init(username: "", password: "")
+    }
```
Comment (yugantarjain) [inline mentorship ios/ViewModels/LoginViewModel.swift:37]: A simple unit test for this method inside the LoginTests class will be nice.

**Review by yugantarjain — CHANGES_REQUESTED** [review]

**Comment by sunjunkie** [comment]
@iMostfa Are you planning to make the changes requested by Yugantar?

**Comment by iMostfa** [comment]
@sunjunkie yes, sure

File: mentorship ios/ViewModels/LoginViewModel.swift (hunk 2 above)
Comment (iMostfa) [reply mentorship ios/ViewModels/LoginViewModel.swift:37]: resolved and added a test case in the main branch.

**Comment by iMostfa** [comment]
@sunjunkie 
@yugantarjain 
added a test case, and edited the comment.

File: mentorship ios/Views/Members/SendRequest.swift [hunk 3]
```diff
@@ -36,6 +36,7 @@ struct SendRequest: View {
         membersService.sendRequest(menteeID: menteeID, mentorID: mentorID, endDate: endDateTimestamp, notes: notes) { response in
             self.membersViewModel.inActivity = false
             self.membersViewModel.sendRequestResponseData = response
+
         }
     }
```
Comment (yugantarjain) [inline mentorship ios/Views/Members/SendRequest.swift]: This file has probably been edited inadvertently...

File: mentorship iosTests/LoginTests.swift [hunk 4]
```diff
@@ -110,6 +110,25 @@ class LoginTests: XCTestCase {
...
+        loginVM.resetLogin()
+
+        let currentLoginData = loginVM.loginData
+
+        //Test
+        XCTAssertEqual("", currentLoginData.username)
+        XCTAssertEqual("", currentLoginData.password)
+
+
+
```
Comment (yugantarjain) [inline mentorship iosTests/LoginTests.swift]: Since we're making changes in any case, can we remove these extra line breaks for neatness...

**Comment by yugantarjain** [comment]
Hi @iMostfa, thanks a lot for working on this issue and the neat solution!

1. I have a couple of very small new comments, please have a look.
2. I can see changes in the LaunchScreen.storyboard file, are they required?
3. You probably also need to pull the latest changes from the upstream develop branch

**Comment by sunjunkie** [comment]
Regarding 2, I think this is from Xcode and not something we need to worry about.

**Comment by yugantarjain** [comment]
Hi @iMostfa, any updates?

File: mentorship ios/Views/Members/SendRequest.swift (hunk 3 above)
Comment (iMostfa) [reply mentorship ios/Views/Members/SendRequest.swift]: yeah my bad, it's solved now

File: mentorship iosTests/LoginTests.swift (hunk 4 above)
Comment (iMostfa) [reply mentorship iosTests/LoginTests.swift]: done

**Comment by iMostfa** [comment]
@yugantarjain I'm very sorry for being late, I've made the changed required, hopes everything is okay this time

**Comment by yugantarjain** [comment]
> @yugantarjain I'm very sorry for being late, I've made the changed required, hopes everything is okay this time

No problem. Amazing work! I'm approving this, thank you!!

**Review by yugantarjain — APPROVED** [review]

**Review by sunjunkie — APPROVED** [review]

**Pull request merged**